src/local-development/align-components-mapping.ts

```diff
@@ -209,17 +259,45 @@ export async function alignComponentsMapping(
 
 				let selectedLocalComponentId: string | symbol | null;
 				switch (newRemoteComponentResolution) {
-					case 'askUser':
-						selectedLocalComponentId = await askForSelectMappedComponent(
-							'remote',
-							remoteOnlyComponent.componentType,
-							remoteOnlyComponent.componentName,
-							remoteOnlyComponent.componentMetadata.label,
-							unlinkedComponents,
-						);
+					case 'askUser': {
+						const userAnswer =
+							userPreferedResolutionOfUnmappedRemote && unlinkedComponents.length === 0
+								? // Use prefered solution if no unlinked components left only
+								  userPreferedResolutionOfUnmappedRemote
+								: // Ask user for the his preferences in all other cases
+								  await askForSelectMappedComponent(
+										'remote',
+										remoteOnlyComponent.componentType,
+										remoteOnlyComponent.componentName,
+										remoteOnlyComponent.componentMetadata.label,
+										unlinkedComponents,
+								  );
+						if (typeof userAnswer === 'string') {
+							actionToProcess = { name: 'mapWith', conterpartyComponentID: userAnswer };
+						} else {
+							switch (userAnswer) {
+								case specialAnswers.CREATE_NEW_COMPONENT:
+									actionToProcess = { name: 'cloneAsNew' };
+									break;
+								case specialAnswers.CREATE_NEW_COMPONENT__FOR_ALL:
+									userPreferedResolutionOfUnmappedRemote = specialAnswers.CREATE_NEW_COMPONENT;
+									actionToProcess = { name: 'cloneAsNew' };
+									break;
+								case specialAnswers.MAP_WITH_NULL:
+									actionToProcess = { name: 'mapWith', conterpartyComponentID: null };
+									break;
+								case specialAnswers.MAP_WITH_NULL__FOR_ALL:
+									userPreferedResolutionOfUnmappedRemote = specialAnswers.MAP_WITH_NULL;
+									actionToProcess = { name: 'mapWith', conterpartyComponentID: null };
+									break;
+								default:
+									throw new Error(`Unknown special answer "${userAnswer?.description}"`);
+							}
+						}
 						break;
+					}
 					case 'cloneAsNew':
-						selectedLocalComponentId = specialAnswers.CREATE_NEW_COMPONENT;
+						actionToProcess = { name: 'cloneAsNew' };
```

Comment:
Should use some constant for option names 'cloneAsNew' is copy pasted too many times.

Reply:
I am not a fan of `const CLONE_AS_NEW = 'cloneAsNew';` in this specific case. There is no special additional value.